src/match.js

```diff
@@ -1,19 +1,25 @@
-var invoker = require('./invoker');
+var _curry2 = require('./internal/_curry2');
+var defaultTo = require('./defaultTo');
 
 
 /**
- * Tests a regular expression against a String
+ * Tests a regular expression against a String. Note that this function
+ * will return an empty array when there are no matches. This differs
+ * from the standard JavaScript string function which returns null when
+ * there are no matches.
  *
  * @func
  * @memberOf R
  * @category String
- * @sig RegExp -> String -> [String] | null
+ * @sig RegExp -> String -> [String]
```

Comment:
> What does that result really mean? Is it an option to filter out undefined values?

It means that your subgroup did not match, eg

    'bye alice'.match(/(good)?bye (alice|bob)/)
    [ 'bye alice', undefined, 'alice' ]